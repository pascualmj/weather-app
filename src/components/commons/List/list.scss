.list {
  display: flex;
  flex-direction: column;

  & > li {
    margin-bottom: 2.2rem;
  }
}
