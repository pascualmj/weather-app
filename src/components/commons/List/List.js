import React from "react";
import PropTypes from "prop-types";
import "./list.scss";

import { renderListDivider } from "../../../functions";

const List = ({ items, renderItems, keyExtractor, divider = null }) => {
  return (
    <ul className="list">
      {items.map((item, index) => (
        <li key={keyExtractor(item)}>
          {renderItems(item)}
          {renderListDivider(index, items.length, divider)}
        </li>
      ))}
    </ul>
  );
};

List.propTypes = {
  items: PropTypes.array.isRequired,
  renderItems: PropTypes.func.isRequired,
  keyExtractor: PropTypes.func.isRequired,
  divider: PropTypes.func,
};

export default List;
