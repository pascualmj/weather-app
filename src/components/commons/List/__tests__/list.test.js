import React from "react";
import { render, getByAltText } from "@testing-library/react";
import "@testing-library/jest-dom";
import List from "../List";

describe("Component: List", () => {
  const setupProps = (items) => ({
    items: [...items],
    keyExtractor: (item) => item,
    renderItems: (item) => item,
  });

  test("Should match snapshot.", () => {
    const { asFragment } = render(<List {...setupProps(["a", "b", "c"])} />);

    expect(asFragment()).toMatchSnapshot();
  });

  test("Should render the items.", () => {
    const { getByText } = render(<List {...setupProps(["hello", "world"])} />);

    getByText("hello");
    getByText("world");
  });
});
