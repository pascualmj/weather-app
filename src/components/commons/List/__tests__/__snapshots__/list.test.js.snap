// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: List Should match snapshot. 1`] = `
<DocumentFragment>
  <ul
    class="list"
  >
    <li>
      a
    </li>
    <li>
      b
    </li>
    <li>
      c
    </li>
  </ul>
</DocumentFragment>
`;
